internal/services/controllers/workflows/queue.go

```diff
@@ -603,79 +603,107 @@ func (wc *WorkflowsControllerImpl) getLock(key string) *sync.Mutex {
 	return actual.(*sync.Mutex)
 }
 
-func (wc *WorkflowsControllerImpl) queueByCancelInProgress(ctx context.Context, tenantId, groupKey string, workflowVersion *dbsqlc.GetWorkflowVersionForEngineRow) error {
+func (wc *WorkflowsControllerImpl) queueByCancelInProgress(ctx context.Context, tenantId string, workflowVersion *dbsqlc.GetWorkflowVersionForEngineRow) error {
 	ctx, span := telemetry.NewSpan(ctx, "queue-by-cancel-in-progress")
 	defer span.End()
 
-	wc.l.Info().Msgf("handling queue with strategy CANCEL_IN_PROGRESS for %s", groupKey)
+	workflowId := sqlchelpers.UUIDToStr(workflowVersion.WorkflowVersion.WorkflowId)
+	maxRuns := int(workflowVersion.ConcurrencyMaxRuns.Int32)
 
-	mutex := wc.getLock(fmt.Sprintf("%s:%s", tenantId, groupKey))
+	mutex := wc.getLock(fmt.Sprintf("%s:%s", tenantId, workflowId))
```

Comment:
Do we still need the mutex?

Reply:
I ended up removing the mutex, good catch